### PR TITLE
[gui-tests] Add oauth2 login test scenario

### PR DESCRIPTION
**Pull request by saw-jan**
Added a test scenario to test the successful oauth2 login (only for oC10 server)

### Related Issues
- part of https://github.com/owncloud/QA/issues/789

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests-oc10</strong> https://drone.owncloud.com/owncloud/client/14321/6/1

:boom: The GUI tests failed.

GUI Logs: https://cache.owncloud.com/public/owncloud/client/14321/oc10/guiReportUpload/index.html

<details>

Server Logs: https://cache.owncloud.com/public/owncloud/client/14321/oc10/guiReportUpload/serverlog.log

Screenshots:
- [login_with_oauth2_enabled.png](https://cache.owncloud.com/public/owncloud/client/14321/oc10/guiReportUpload/screenshots/login_with_oauth2_enabled.png)

</details>

File: test/gui/shared/scripts/pageObjects/AccountConnectionWizard.py [hunk 1]
```diff
@@ -183,13 +183,20 @@ def basicLogin(username, password):
...
     @staticmethod
     def oidcLogin(username, password):
+        AccountConnectionWizard.browser_login(username, password, 'oidc')
+
+    @staticmethod
+    def oauthLogin(username, password):
+        AccountConnectionWizard.browser_login(username, password, 'oauth')
+
+    @staticmethod
+    def browser_login(username, password, login_type=None):
```
Comment (individual-it) [inline test/gui/shared/scripts/pageObjects/AccountConnectionWizard.py:193]: I thought we are using :camel: - case function names
```suggestion
    def browserLogin(username, password, login_type=None):
```
same for `authorize_via_webui`, but that can be fixed in a separate PR

**Review by individual-it — CHANGES_REQUESTED** [review]

File: test/gui/webUI/login.spec.js [hunk 2]
```diff
@@ -1,32 +1,54 @@
...
+test("oC10 login @oauth", async ({ page }) => {
+  // login
+  await page.fill("#user", config.username);
+  await page.fill("#password", config.password);
+  await page.click("button[type=submit]");
+  // authorize
+  await page.click("button >> text=Authorize");
+  // confirm successful login
+  await expect(page.locator("span.error")).toContainText(
+    "The application was authorized successfully"
```
Comment (amrita-shrestha) [inline test/gui/webUI/login.spec.js:52]: do we have to add such normal comments also?

Comment (saw-jan) [reply test/gui/webUI/login.spec.js:52]: just to group the actions.

**Review by individual-it — APPROVED** [review]

**Pull request merged**